tools/building.py

```diff
@@ -1445,9 +1445,9 @@ def get_binaryen_bin():
   return rtn
 
 
-# track the arguments of the last binaryen command that writes an output. this
-# is used for knowing whether we need to strip debug info at the very end.
-last_writing_binaryen_cmd = None
+# track whether the last binaryen command kept debug info around. this is used
+# to see whether we need to do an extra step at the end to strip it.
+binaryen_kept_debug_info = False
```

Comment:
Wouldn't be more accurate to track "has_stripped_debug" .. i.e.:

```
if `-g` not in cmd:
   global binaryen_did_strip
   binaryn_did_strip = True
```

That way we track whether *any* binaryen command did stripping, not just what the last one did.

Reply:
In principle a later command could add the names section back (the names would be numbers, but they'd be there). I don't think we do that for any reason atm, though, that's true.

In any case, the two are equivalent as "stripping" is the same is not adding `-g`, and we constantly do not add `-g` when we don't need it.

Reply:
ping @sbc100